lib/util/config.js

```diff
@@ -0,0 +1,275 @@
+                                        solutionSet.add(
+                                            `Run 'mcdev upgrade' to fix missing or changed configuration options`
+                                        );
+                                        missingFields.push(`${key}.${subkey}.${dkey}`);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        // check if project config version is outdated compared to user's mcdev version
+        if (
+            !properties.version ||
+            (![null, 'patch'].includes(semver.diff(packageJsonMcdev.version, properties.version)) &&
+                semver.gt(packageJsonMcdev.version, properties.version))
+        ) {
+            errorMsgs.push(
+                `Your project's config version ${properties.version} is lower than your Accenture SFMC DevTools version ${packageJsonMcdev.version}`
+            );
+            solutionSet.add(`Run 'mcdev upgrade' to ensure optimal performance`);
+            missingFields.push('version');
+        }
+        if (silent) {
+            const err = new Error('Missing Fields');
+            err.missingFields = missingFields;
+            throw err;
+        } else {
+            if (errorMsgs.length) {
+                const errorMsgOutput = [
+                    `Found problems in your ./${Util.configFileName} that you need to fix first:`,
+                ];
+                for (const msg of errorMsgs) {
+                    errorMsgOutput.push(' - ' + msg);
+                }
+                const errorMsgText = errorMsgOutput.join('\n');
+                if (Util.skipInteraction) {
+                    throw new Error(errorMsgText);
+                }
+                Util.logger.error(errorMsgText);
+                Util.logger.info(
+                    [
+                        'Here is what you can do to fix these issues:',
+                        ...Array.from(solutionSet),
+                    ].join('\n- ')
+                );
+                const responses = await inquirer.prompt([
+                    {
+                        type: 'confirm',
+                        name: 'runUpgradeNow',
+                        message: `Do you want to run 'mcdev upgrade' now?`,
+                        default: true,
+                    },
+                ]);
+                if (responses.runUpgradeNow) {
+                    // use _execSync here to avoid a circular dependency
+                    this.execSync('mcdev', ['upgrade']);
+                }
+                throw new Error(errorMsgOutput);
+            }
+        }
+    },
+    /**
+     * defines how the properties.json should look like
+     * used for creating a template and for checking if variables are set
+     *
+     * @returns {object} default properties
+     */
+    getDefaultProperties: async function () {
```

Comment:
and again the other type of notation.
this sets a property to an anonymous method rather than define the method getDefaultProperties as part of the Config object